experiments/19_rbpf_portfolio.py

```
@@ -0,0 +1,534 @@
+def _portfolio_sharpe(
+    signals_list: list[np.ndarray],
+    returns_list: list[np.ndarray],
+    tc_bps: float,
+) -> float:
+    """Mean individual Sharpe across N contracts (portfolio proxy).
+
+    Different contracts have different bar counts (FX/grains have missing
+    1-min bars during RTH), so we cannot stack them into a single matrix.
+    Instead we compute per-contract annualised Sharpe after transaction costs
+    and return the mean — this equals the portfolio Sharpe in the limit of
+    many uncorrelated strategies (ρ → 0):
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `ρ` (GREEK SMALL LETTER RHO). Did you mean `p` (LATIN SMALL LETTER P)?

(RUF002)